synnax/cmd/start.go

```diff
@@ -431,6 +438,70 @@ func getIntegrations(enabled, disabled []string) []string {
 	return integrations // Ensure a return value in case both slices are empty
 }
 
+// sets the base permissions that need to exist in the server.
+func maybeSetBasePermission(
+	ctx context.Context,
+	db *gorp.DB,
+	rbacSvc *rbac.Service,
+) error {
+	return db.WithTx(ctx, func(tx gorp.Tx) error {
+		// base policies that need to be created
+		basePolicies := map[ontology.Type]action.Action{
+			"label":       action.All,
+			"cluster":     action.All,
+			"channel":     action.All,
+			"node":        action.All,
+			"group":       action.All,
+			"range":       action.All,
+			"range-alias": action.All,
+			"workspace":   action.All,
+			"lineplot":    action.All,
+			"rack":        action.All,
+			"device":      action.All,
+			"task":        action.All,
+			"user":        action.Retrieve,
+			"schematic":   action.Retrieve,
+			"policy":      action.Retrieve,
+			"builtin":     action.Retrieve,
+		}
+		// for migration purposes, some old base policies that need to be deleted
+		oldBasePolicies := map[ontology.Type]action.Action{}
+
+		existingPolicies := make([]rbac.Policy, 0, len(basePolicies))
+		policiesToDelete := make([]uuid.UUID, 0, len(oldBasePolicies))
+		if err := rbacSvc.NewRetriever().WhereSubjects(user.OntologyTypeID).
+			Entries(&existingPolicies).Exec(ctx, tx); err != nil {
+			return err
+		}
+		for _, p := range existingPolicies {
+			if len(p.Subjects) != 1 || len(p.Objects) != 1 || len(p.Actions) != 1 {
```

Comment:
I would add a comment explaining why having a policy with more than 1 object, subject, or action is problematic.